src/components/azure-storage/azureProjectVolume.ts

```diff
@@ -0,0 +1,66 @@
+import { assert } from "@sindresorhus/is";
+import gitlab from "@socialgouv/kosko-charts/environments/gitlab";
+import { updateMetadata } from "@socialgouv/kosko-charts/utils/updateMetadata";
+import { PersistentVolume, PersistentVolumeClaim } from "kubernetes-models/v1";
+
+export function azureProjectVolume(
+  name: string,
+  { storage }: { storage: string }
+): [PersistentVolumeClaim, PersistentVolume] {
+  const globalEnv = gitlab(process.env);
+  assert.nonEmptyObject(globalEnv.labels);
+
+  const application = globalEnv.labels.application;
+  const metadata = {
+    annotations: globalEnv.annotations ?? {},
+    labels: globalEnv.labels,
+    namespace: globalEnv.namespace,
+  };
+  const pv = `${application}-${name}`;
+
+  const persistentVolumeClaim = new PersistentVolumeClaim({
+    metadata: {
+      name,
+    },
+    spec: {
+      accessModes: ["ReadWriteMany"],
+      resources: {
+        requests: {
+          storage,
+        },
+      },
+      selector: {
+        matchLabels: {
+          usage: pv,
+        },
+      },
+      storageClassName: "",
+    },
+  });
+
+  const persistentVolume = new PersistentVolume({
+    metadata: {
+      labels: {
+        usage: pv,
+      },
+      name: pv,
+    },
+    spec: {
+      accessModes: ["ReadWriteMany"],
+      azureFile: {
+        secretName: `azure-${globalEnv.labels.team}-volume`,
+        secretNamespace: globalEnv.namespace.name,
+        shareName: name,
+      },
+      capacity: {
+        storage,
+      },
+      persistentVolumeReclaimPolicy: "Delete",
```

Comment:
should'nt we use "Retain" in preprod/prod ?

Reply:
We could but `Delete` seems to be faster to establish a connection